### PR TITLE
Verify component signature and make install works

### PR DESCRIPTION
**Pull request by july2993**
~~blocked by #244~~
- Verify component signature and test we can load component manifest
- Fix panic to close when enable v1 & platform format

after this pr: `install tikv` can success (enable v1)

test:
- alias tt="TIUP_USE_V1=true TIUP_HOME=/Users/huangjiahao/code/tiup/testprofile TIUP_MIRRORS=/Users/huangjiahao/local-mirrors /Users/huangjiahao/code/tiup/bin/tiup"
- tt install tikv

where mirrors dir contains the manifests in generate by `tools/migrate` and the ..tar.gz of components.
TIUP_HOME contains only the trusted root.json

After running `install tikv` the profile dir:
```
➜  tiup git:(test_com) ✗ find testprofile
testprofile
testprofile/init.sh
testprofile/snapshot.json
testprofile/components
testprofile/components/tikv
testprofile/components/tikv/v4.0.0-rc.2
testprofile/components/tikv/v4.0.0-rc.2/tikv-server
testprofile/timestamp.json
testprofile/index.json
testprofile/root.json
testprofile/tikv.json
```

but 
```
➜  tiup git:(test_com) ✗ tt tikv -h
component `tikv` doesn't install
```
will check this as follow up.

### CODE REVIEW

File: pkg/meta/meta.go [hunk 1]
```diff
@@ -90,6 +90,15 @@ func (env *Environment) Profile() *localdata.Profile {
 	return env.profile
 }
 
+// Close release resource of env.
+func (env *Environment) Close() error {
+	if os.Getenv(EnvNameV1) == "" {
```
Comment (nrc) [inline pkg/meta/meta.go:95]: I think this check should be `env.repo != nil` (once the repos are initialized we should no longer check the env vars in order to make them easy to override, e.g., in testing).

File: pkg/repository/v1_repository.go [hunk 2]
```diff
@@ -420,6 +420,16 @@ func (r *V1Repository) checkTimestamp() (*v1manifest.FileHash, error) {
 	return &hash, r.local.SaveManifest(manifest, v1manifest.ManifestFilenameTimestamp)
 }
 
+func (r *V1Repository) fetchComponentManefest(index *v1manifest.Index, url string, com *v1manifest.Component, maxSize uint) (*v1manifest.Manifest, error) {
```
Comment (nrc) [inline pkg/repository/v1_repository.go:423]:
```suggestion
func (r *V1Repository) fetchComponentManifest(index *v1manifest.Index, url string, com *v1manifest.Component, maxSize uint) (*v1manifest.Manifest, error) {
```

File: pkg/repository/v1manifest/types.go [hunk 3]
```diff
@@ -171,6 +186,21 @@ func (manifest *Root) GetRootKeyStore() (ks crypto.KeyStore, err error) {
 	return
 }
 
+// GetComponentKeys the the threshold and keys for the specified component id.
```
Comment (nrc) [inline pkg/repository/v1manifest/types.go:189]:
```suggestion
// GetComponentKeys returns the threshold and keys for the specified component id.
```

File: pkg/repository/v1manifest/manifest.go [hunk 4]
```diff
@@ -422,6 +459,8 @@ func ReadManifest(input io.Reader, role ValidManifest, root *Root) (*Manifest, e
 		if err != nil {
 			return nil, err
 		}
+	} else {
+		return nil, errors.Errorf("unknow type: %s", role.Base().Ty)
```
Comment (nrc) [inline pkg/repository/v1manifest/manifest.go:463]:
```suggestion
		return nil, errors.Errorf("unknown type: %s", role.Base().Ty)
```

File: pkg/repository/v1manifest/local_manifests.go [hunk 5]
```diff
@@ -91,13 +91,20 @@ func (ms *FsManifests) LoadManifest(role ValidManifest) (bool, error) {
 }
 
 // LoadComponentManifest implements LocalManifests.
-func (ms *FsManifests) LoadComponentManifest(filename string) (*Component, error) {
-	var role Component
-	exists, err := ms.load(filename, &role)
-	if !exists {
-		return nil, nil
+func (ms *FsManifests) LoadComponentManifest(index *Index, filename string) (*Component, error) {
+	com := new(Component)
```
Comment (nrc) [inline pkg/repository/v1manifest/local_manifests.go:95]: Could you continue to use `load` and move the call to `ReadManifest` into `LoadManifest`?

**Comment by july2993** [comment]
@nrc PTAL

**Pull request merged**